### PR TITLE
[incubator-kie-drools#5679] New DRL parser : Test failure : RHS end w…

### PR DESCRIPTION
**Pull request by tkobayas**
…ithout preceding white-space

**Issue:** https://github.com/apache/incubator-kie-drools/issues/5679

### CODE REVIEW

**Comment by tkobayas** [comment]
For `drools-model-codegen` tests, 

Before the fix
```
[ERROR] Tests run: 2433, Failures: 592, Errors: 16, Skipped: 9
```

After the fix
```
[ERROR] Tests run: 2433, Failures: 435, Errors: 14, Skipped: 9
```

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLLexer.g4 [hunk 1]
```diff
@@ -158,4 +158,9 @@ DrlUnicodeEscape
 mode RHS;
 RHS_WS : [ \t\r\n\u000C]+ -> channel(HIDDEN);
 DRL_RHS_END : 'end' [ \t]* SEMI? [ \t]* ('\n' | '\r\n' | EOF) {setText("end");} -> popMode;
-RHS_CHUNK : ~[ \t\r\n\u000C]+ ;
+RHS_CHUNK
+    : ~[ ;})\t\r\n\u000C]+ [;})]? // ; } ) could be a delimitter proceding 'end'
+    | SEMI
+    | RBRACE
+    | RPAREN
+    ;
```
Comment (tkobayas) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLLexer.g4:166]: Not elegant definition, but `RHS_CHUNK` is just used to let the Lexer find `end`. In `DRLVisitorImpl`, the whole RHS is eventually set as String.

**Comment by tkobayas** [comment]
@mariofusco @yurloc Please review, thanks!

**Review by mariofusco — APPROVED** [review]

**Comment by yurloc** [comment]
LGTM!

**Comment by yurloc** [comment]
There were 557 failed tests (-163).

**Pull request merged**